.github/workflows/tools_ci.yml

```diff
@@ -341,3 +341,34 @@ jobs:
           workflow_file_name: containers.yml
           client_payload: '{"sc-ref": "${{ github.event.pull_request.head.ref || github.sha }}"}'
           wait_interval: 60
+
+  lint_tcl:
+    name: Lint OpenROAD TCL scripts
+    if: always()
+    needs: [build_tool_builder, build_sc_tools]
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ needs.build_tool_builder.outputs.sc_tools }}
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ github.token }}
+    env:
+      COMMAND_SPEC_PATH: tclint-openroad-spec.json
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v4
+
+      - name: Install tclint
+        run: |
+          python3 -m pip install tclint -c requirements.txt
+
+      - name: Generate OpenROAD command spec
+        run: |
+          tclint-plugins make-spec openroad --output $COMMAND_SPEC_PATH
```

Comment:
It might be nice if this was generated by the openroad build image (that would require the command-spec.json format to be fairly stable though